### ISSUE TITLE
Training on your own dataset

### ISSUE THREAD

**Issue by eyildiz-ugoe** [Microsoft/singleshotpose#17]
One other impressive thing which could really help people reproduce the results is to enable a way that can guide them to train the network from the scratch **using their own dataset**. Then my question would be the following:

Would this mean the users who want to give it a try have to have:

1. The JPEG training images (rgb) and the ground truth (binary) masks of the objects, organized as neatly as the objects now in the dataset such as ape, benchvise, cat, etc. (similar folder structure obviously)

2. The necessary lists for training and testing (train.txt, test,txt)

3. The 3D model of the object (.ply)

Would they need something else to get the system running on their own training dataset? I see something called "training range" which I am not sure what for, but I think you can answer the question better than I speculate, hence the reason why I opened this issue.

Could you perhaps write a small guide for us so that we could utilize the work on our own dataset? To see how it would perform in other users' own domains is something one has to consider, naturally. This would also make the paper cited by way more people if they could just try and see how it performs on their own dataset.

**Comment by btekin** [comment]
To train on your own dataset, you just need to have the same folder structure for the dataset. 

As you have already written, this would mean to have separate files/subfolders for 

1) JPEG images and labels (and optionally "masks" if you want to change the background of your training images to be more robust to diverse backgrounds). 

2) train.txt and test.txt, and 

3) 3D model of the object (.ply)

neatly organized in your dataset folder. Training range is just to specify the which training images to use. We use it to have exactly the same training splits with previous work on object pose estimation (BB8 paper by Rad & Lepetit, ICCV'17) Since we already have train.txt, it is not really needed though, I'll remove it in the next version. I'll also try to write a small guide in README.md putting together what needs to be done to train on your own dataset.

**Comment by btekin** [comment]
I have just added a description on how to train on a custom dataset.

**Comment by eyildiz-ugoe** [comment]
Thank you for the description. :)
